CMakeLists.txt

```diff
@@ -2,6 +2,12 @@ cmake_minimum_required(VERSION 3.12)
 project(metalog)
 
 set(CMAKE_CXX_STANDARD 17)
-include_directories(${PROJECT_SOURCE_DIR})
 
-add_executable(metalog main.cc)
\ No newline at end of file
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}/bin)
+
+include_directories(include)
+
+add_executable(metalog example/metalog.cc)
+
+install(DIRECTORY include/ DESTINATION include)
```

Comment:
I don't think wrapping the headers into an "include" dir for a head-only library is necessary.

maybe use  ` ${CMAKE_SOURCE_DIR}/metalog` is a better choice

Reply:
https://github.com/avplayer/avhttp/blob/master/CMakeLists.txt#L78